### ISSUE TITLE
CircleCI failed due to react-native-reanimated (android)

### ISSUE THREAD

**Issue by mahipalsingh-syt** [software-mansion/react-native-reanimated#3513]
### Description

Circle CI lane failed at react-native-reanimated in android

log added

```Starting a Gradle Daemon (subsequent builds will be faster)

> Configure project :react-native-reanimated
Native libs debug enabled: false
Android gradle plugin: 7.0.3
Gradle: 7.3
building Reanimated2
Checking the license for package NDK (Side by side) 21.4.7075529 in /opt/android/sdk/licenses
License for package NDK (Side by side) 21.4.7075529 accepted.
Preparing "Install NDK (Side by side) 21.4.7075529 (revision: 21.4.7075529)".
"Install NDK (Side by side) 21.4.7075529 (revision: 21.4.7075529)" ready.
Installing NDK (Side by side) 21.4.7075529 in /opt/android/sdk/ndk/21.4.7075529
"Install NDK (Side by side) 21.4.7075529 (revision: 21.4.7075529)" complete.
"Install NDK (Side by side) 21.4.7075529 (revision: 21.4.7075529)" finished.
[CXX1300] CMake '3.18.1' was not found in SDK, PATH, or by cmake.dir property.
[CXX1300] CMake '3.18.1' was not found in SDK, PATH, or by cmake.dir property.

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':react-native-reanimated'.
> [CXX1300] CMake '3.18.1' was not found in SDK, PATH, or by cmake.dir property.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org
```

### Steps to reproduce

Run CircleCI lane in reactnative for android 

### Snack or a link to a repository

https://snack.expo.io/

### Reanimated version

2.10.0

### React Native version

0.68.0

### Platforms

Android

### Workflow

React Native (without Expo)

### Architecture

Paper (Old Architecture)

### Build type

Release mode

### Acknowledgements

Yes

**Comment by tomekzaw** [comment]
Hey @mahipalsingh-syt!

Reanimated 2.10 requires CMake 3.12+ (see [here](https://github.com/software-mansion/react-native-reanimated/blob/main/android/CMakeLists.txt#L2)).

Make sure that it's installed and available when building app for Android.

**Comment by carestad** [comment]
Seeing this too. Why does it _only_ work with CMake 3.18.1 and not newer versions of CMake as well? I have 3.22.1 installed and still get this error.

This wasn't an issue before v2.10.0, so rolling back to v2.9.1 fixed this for me.

**Comment by tomekzaw** [comment]
@carestad There may be another library that sets the maximum supported CMake version and that's why `CMake '3.22.1' found in PATH did not satisfy requested version.`. Could you please verify this hypothesis on your setup? 🙏

**Comment by carestad** [comment]
Isn't that a bit weird, since the only dependency that now have changed is `react-native-reanimated`, since I rolled it back to v2.9.1, and building the app now works again?

But I could of course try and figure out if there could be another dependency in play here. Is there anything in particular I could search for or grep for? And if so, in what folder?

**Comment by tomekzaw** [comment]
Yeah, I'm thinking there might be `react-native-something` that requires CMake let's say `<=3.11`. With Reanimated 2.9.1 we required `>=3.5.1` so i.e. `3.10.2` was fine. With 2.10 we require `>=3.12` so there is no version that satisfies both conditions.

Instead of downgrading to 2.9.1, can you install CMake 3.18.1 with
```$ANDROID_HOME/cmdline-tools/latest/bin/sdkmanager "cmake;3.18.1"```
and use Reanimated 2.10?

**Comment by tomekzaw** [comment]
@carestad Great news!

If anyone else encounters this issue (CMake 3.18 not found), another approach might be to remove this line in `node_modules/react-native-reanimated/android/build.gradle`:
https://github.com/software-mansion/react-native-reanimated/blob/acf7429068c4f801eef41267c2a94b5515ec4ff8/android/build.gradle#L284

Let us know if it works for you.

**Comment by billnbell** [comment]
How do we move that?

```
    externalNativeBuild {
        cmake {
            path "CMakeLists.txt"
            version "3.18.1"
        }
    }
```

to?

```
    externalNativeBuild {
        cmake {
            path "CMakeLists.txt"
        }
    }
```

**Comment by mym0404** [comment]
> Hey @mahipalsingh-syt!
> 
> Reanimated 2.10 requires CMake 3.12+ (see [here](https://github.com/software-mansion/react-native-reanimated/blob/main/android/CMakeLists.txt#L2)).
> 
> Make sure that it's installed and available when building app for Android.
> 
> Here's how you can install CMake 3.18.1:
> 
> ```
> $ANDROID_HOME/cmdline-tools/latest/bin/sdkmanager "cmake;3.18.1"
> ```

This resolved my case.

**Comment by SumitRajpal** [comment]
For my native project - **3.22.1 failed** react-native-reanimated specifically requires **3.18.1** ,
uninstalling 3.22.1 and reinstalling 3.18.1 works like a charm 
![image](https://user-images.githubusercontent.com/7436825/187846655-ffca9739-67f0-4c92-ab82-7e76f747bd7c.png)

**Comment by sanaspra** [comment]
> For my native project - **3.22.1 failed** react-native-reanimated specifically requires **3.18.1** , uninstalling 3.22.1 and reinstalling 3.18.1 works like a charm

How did you install the exact version?

**Comment by SumitRajpal** [comment]
> 

![image](https://user-images.githubusercontent.com/7436825/187905465-5696dfd9-6cea-4e01-b9eb-71e48a24ac3e.png)

**Comment by hosseintalebi** [comment]
@tomekzaw thanks for the fix in https://github.com/software-mansion/react-native-reanimated/pull/3521. Do you know when we can expect a new version with your fix?

**Comment by temchannat** [comment]
> > 
> 
> ![image](https://user-images.githubusercontent.com/7436825/187905465-5696dfd9-6cea-4e01-b9eb-71e48a24ac3e.png)

Thank you so much. This works for me.

**Comment by tomekzaw** [comment]
Hey, just wanted to let you know that this issue has been fixed in 3.0.0-rc.3.

For Reanimated 2 users, we plan to release 2.11 next week.

**Comment by bradleyflood** [comment]
Thank you @tomekzaw. Appreciate all your hard work!

**Comment by mantegnous** [comment]
I have the same problem with reanimated 2.8.0 ... how can resolve? 

install-missing-android-tools  3.1.0

reanimated find '3.24.1' version of cmake

**Comment by jobayer977** [comment]
react-native-reanimated@2.9.1

This version works for me

**Comment by janpe** [comment]
> Hey, just wanted to let you know that this issue has been fixed in 3.0.0-rc.3.
> 
> For Reanimated 2 users, we plan to release 2.11 next week.

Hi! Any new on 2.11?

**Comment by baha-udin** [comment]
https://github.com/software-mansion/react-native-reanimated/issues/3513#issuecomment-1228568228

for some reason, recommend to install cmake version 3.18.1 for stable upcoming project, until react-native-reanimated update to solve this problem

**Comment by zorand1986** [comment]
@tomekzaw Any info on the 2.11.0? When can we expect it? Thanks 🙏

**Comment by tomekzaw** [comment]
@zorand1986 We have just published 2.11.0: https://www.npmjs.com/package/react-native-reanimated/v/2.11.0

**Comment by janpe** [comment]
@tomekzaw is this patched in 2.11.0? https://github.com/advisories/GHSA-2j79-8pqc-r7x6

Patched versions only lists v3

**Comment by tomekzaw** [comment]
@janpe Yes, it was already patched in 2.10.0.
